streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/processors/FirstProcessor.kt

```diff
@@ -11,13 +12,20 @@ class FirstProcessor<T>(parentPublisher: Publisher<T>) :
     }
 
     class FirstProcessorSubscription<T>(s: Subscriber<in T>) : ProcessorSubscription<T, T>(s) {
-
+        private val hasReceivedValue = AtomicReference(false)
         override fun onNext(t: T, subscriber: Subscriber<in T>) {
             if (t != null) {
+                hasReceivedValue.compareAndSet(false, true)
                 cancelActiveSubscription()
                 subscriber.onNext(t)
                 subscriber.onComplete()
             }
         }
+
+        override fun onComplete() {
+            if (!hasReceivedValue.value) {
```

Comment:
How can a `first()` processor receive a `onComplete()` signal after cancelling it's active subscription? I'm expecting the `cancelActiveSubscription()` to avoid `onComplete()` to be called afterwards, no?